customers-service/src/main/java/com/petclinic/customers/businesslayer/PhotoServiceImpl.java

```diff
@@ -0,0 +1,97 @@
+package com.petclinic.customers.businesslayer;
+
+import com.petclinic.customers.customerExceptions.exceptions.NotFoundException;
+import com.petclinic.customers.datalayer.*;
+import com.petclinic.customers.presentationlayer.PetRequest;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class PhotoServiceImpl implements PhotoService{
+    private final PhotoRepository photoRepository;
+    private final OwnerRepository ownerRepository;
+    private final PetRepository petRepository;
+
+    public PhotoServiceImpl(PhotoRepository photoRepository, OwnerRepository ownerRepository, PetRepository petRepository) {
+        this.petRepository = petRepository;
+        this.ownerRepository = ownerRepository ;
+        this.photoRepository = photoRepository;
+    }
+
+    @Override
+    public String setPhotoOwner(Photo photo, int ownerId) {
+        try {
+        photoRepository.save(photo);
+        Owner owner = ownerRepository.findOwnerById(ownerId);
+        int deleteId = owner.getImageId();
+        //find by name needs to be reworked, this is a bad idea but will be changed when migrated to reactive
+        owner.setImageId(photoRepository.findPhotoByName(photo.getName()).getId());
+        ownerRepository.save(owner);
+        if(deleteId!=1){
+            this.deletePhoto(deleteId);
+        }
```

Comment:
I think extracting the predicate to a variable would improve the readability, what do you think about this change?

```suggestion
        boolean shouldPhotoBeDeleted = deleteId!=1;
        if(shouldPhotoBeDeleted){
            this.deletePhoto(deleteId);
        }
```